src/storage/patch/SparqlUpdatePatchHandler.ts

```diff
@@ -73,6 +73,7 @@ export class SparqlUpdatePatchHandler extends PatchHandler {
     const def = defaultGraph();
     const deletes = op.delete ?? [];
     const inserts = op.insert ?? [];
+    const where: BaseQuad[] = typeof op.where !== 'undefined' ? op.where.patterns : [];
```

Comment:
This is not going to work in the cases when the `where` body is more complex and does not have a bgp at the root, such as a union for example. There needs to be a check here to see if the `where` body is a bgp if it exists (and assume we don't support it in case it is not a bgp). You can do the same thing as the `isDeleteInsert` and `isComposite` functions above.

Reply:
Would it be something like the new version?